doc/src/manual/variables-and-scoping.md

```diff
@@ -93,244 +102,356 @@ Note that the interactive prompt (aka REPL) is in the global scope of the module
+scope rule applies and `x` is created as local to the for loop and therefore global `x` remains
+undefined after the loop executes. Next, let's consider the body of `sum_to′` extracted into global
+scope, fixing `n = 10`:
 
-julia> x, y # verify that global x and y are unchanged
-(1, 2)
 ```
-
-The reason to allow modifying local variables of parent scopes in
-nested functions is to allow constructing [`closures`](https://en.wikipedia.org/wiki/Closure_%28computer_programming%29)
-which have private state, for instance the `state` variable in the
-following example:
+s = 0
+for i = 1:10
+    t = s + i
+    s = t
+end
+s
+@isdefined(t)
+ ```
+
+What does this code do? Hint: it's a trick question. The answer is "it depends." If this code is
+entered interactively, it behaves the same way it does in a function body. But if the code appears
+in a file, it  prints an ambiguity warning and throws and undefined variable error. Let's see it
+working in the REPL first:
 
 ```jldoctest
-julia> let state = 0
-           global counter() = (state += 1)
-       end;
+julia> s = 0 # global
+0
 
-julia> counter()
-1
+julia> for i = 1:10
+           t = s + i # new local `t`
+           s = t # assign global `s`
+       end
 
-julia> counter()
-2
+julia> s # global
+55
+
+julia> @isdefined(t) # global
+false
 ```
 
-See also the closures in the examples in the next two sections. A variable,
-such as `x` in the first example and `state` in the second, that is inherited
-from the enclosing scope by the inner function is sometimes called a
-*captured* variable. Captured variables can present [performance challenges
-discussed in performance tips](@ref man-performance-captured).
+The REPL approximates being in the body of a function by deciding whether assignment inside the loop
+assigns to a global  or creates new local based on whether a global variable by that name is defined
+or not. If a global by the name exists, then the assignment updates it. If no global exists, then
+the assignment creates a new local variable. In this example we see both cases in action:
+
+* There is no global named `t`, so `t = s + i` creates a new `t` that is local to the for loop;
+* There is a global named `s`, so `s = t` assigns to it.
 
-The distinction between inheriting global scope and nesting local scope
-can lead to some slight differences between functions
-defined in local versus global scopes for variable assignments.
-Consider the modification of the last example by moving `bar` to the global scope:
+The second fact is why execution of the loop changes the global value of `s` and the first fact is
+why `t` is still undefined after the loop executes. Now, let's try evaluating this same code as
+though it were in a file instead:
 
 ```jldoctest
-julia> x, y = 1, 2;
+julia> code = """
+       s = 0 # global
+       for i = 1:10
+           t = s + i # new local `t`
+           s = t # new local `s` with warning
+       end
+       s # global
+       @isdefined(t) # global
```

Comment:
Maybe this should be `s, @isdefined(t)`, since otherwise only the second result would ever be visible.